### PR TITLE
Port to the new subclassing infrastructure from the glib bindings

### PR DESCRIPTION
**Pull request by sdroege**
This currently is patched to use my personal glib branch.

See https://github.com/gtk-rs/glib/pull/392

### CODE REVIEW

**Comment by EPashkin** [comment]
@sdroege Thanks, now it looks much nicer
Please add "glib/subclassing" to commented default

File: src/bin/listbox_model.rs [hunk 1]
```diff
@@ -344,61 +343,41 @@ mod row_data {
...
-            fn get_property(&self, _obj: &glib::Object, id: u32) -> Result<glib::Value, ()> {
-                let prop = &PROPERTIES[id as usize];
+            fn get_property(&self, _obj: &glib::Object, id: usize) -> Result<glib::Value, ()> {
+                let prop = &PROPERTIES[id];
 
                 match *prop {
-                    Property::String("name", ..) => Ok(self.name.borrow().clone().to_value()),
-                    Property::UInt("count", ..) => Ok(self.count.borrow().clone().to_value()),
+                    subclass::Property("name", ..) => Ok(self.name.borrow().clone().to_value()),
+                    subclass::Property("count", ..) => Ok(self.count.borrow().clone().to_value()),
```
Comment (EPashkin) [inline src/bin/listbox_model.rs]: .clone() seems unneeded for both

**Comment by sdroege** [comment]
>  Please add "glib/subclassing" to commented default

Done and all updated. CI will fail now until https://github.com/gtk-rs/glib/pull/392 is merged

File: src/bin/listbox_model.rs [hunk 2]
```diff
@@ -409,17 +388,14 @@ mod row_data {
...
-                    ])
-                    .unwrap()
-                    .downcast_unchecked()
-            }
+            glib::Object::new(
+                Self::static_type(),
+                &[("name", &name),
+                  ("count", &count),
+                ])
+                .unwrap()
```
Comment (GuillaumeGomez) [inline src/bin/listbox_model.rs]: Please use `expect`.

Comment (sdroege) [reply src/bin/listbox_model.rs]: Fixed, thanks!

**Comment by sdroege** [comment]
@GuillaumeGomez all green :)

**Comment by GuillaumeGomez** [comment]
Thanks!

**Pull request merged**